src/api/network/asio/Imp.cpp

```diff
@@ -290,27 +270,25 @@ auto Asio::Imp::process_address_query(
     if (!promise) { return; }
 
     try {
-        const auto string = [&] {
-            auto output = CString{};
-            const auto body = future.get().body();
-
-            switch (type) {
-                case ResponseType::IPvonly: {
-                    auto parts = Vector<CString>{};
-                    algo::split(parts, body, algo::is_any_of(","));
-
-                    if (parts.size() > 1) { output = parts[1]; }
-                } break;
-                case ResponseType::AddressOnly: {
-                    output = body;
-                } break;
-                default: {
-                    throw std::runtime_error{"Unknown response type"};
-                }
-            }
+        CString string{};
+        const auto body = future.get().body();
+
+        switch (type) {
+            case ResponseType::IPvonly: {
+                auto parts = Vector<CString>{};
+                algo::split(parts, body, algo::is_any_of(","));
+
+                if (parts.size() > 1) { string = parts[1]; }
+            } break;
 
-            return output;
-        }();
+            case ResponseType::AddressOnly: {
+                string = body;
+            } break;
+
+            default: {
+                throw std::runtime_error{"Unknown response type"};
```

Comment:
Is it enough information in an exception? will it be easy tracebable by greping the code ? maybe some identifier of exception or method name could help that or some specific detail ?

Reply:
The signature should be changed to noexcept(false) in the first place, or there won't be any useful exception thrown.